airbyte-integrations/connectors/destination-iceberg/src/main/java/io/airbyte/integrations/destination/iceberg/config/catalog/NessieCatalogConfig.java

```diff
@@ -0,0 +1,98 @@
+  private final String token;
+
+  public NessieCatalogConfig(@NotNull JsonNode catalogConfig) {
+    Preconditions.checkArgument(null != catalogConfig.get(NESSIE_CATALOG_URI_CONFIG_KEY), "%s is required", NESSIE_CATALOG_URI_CONFIG_KEY);
+    JsonNode tokenNode = catalogConfig.get(NESSIE_CATALOG_TOKEN_CONFIG_KEY);
+    JsonNode clientApiNode = catalogConfig.get(NESSIE_CLIENT_API_VERSION_KEY);
+    JsonNode refNode = catalogConfig.get(NESSIE_CATALOG_REFERENCE_KEY);
+    JsonNode authTypeNode = catalogConfig.get(NESSIE_CATALOG_AUTHENTICATION_TYPE_KEY);
+    this.token = null != tokenNode ? tokenNode.asText() : null;
+    this.nessieApiVersion = null != clientApiNode ? clientApiNode.asText() : null;
+    this.uri = catalogConfig.get(NESSIE_CATALOG_URI_CONFIG_KEY).asText();
+    this.ref = null != refNode ? refNode.asText() : null;
+    this.authenticationType = null != authTypeNode ? authTypeNode.asText() : null;;
+  }
+
+  @Override
+  public Map<String, String> sparkConfigMap() {
+    Map<String, String> configMap = new HashMap<>();
+    configMap.put("spark.network.timeout", "300000");
+    configMap.put("spark.sql.defaultCatalog", CATALOG_NAME);
+    configMap.put("spark.sql.extensions", "org.apache.iceberg.spark.extensions.IcebergSparkSessionExtensions,org.projectnessie.spark.extensions.NessieSparkSessionExtensions");
+    configMap.put("spark.sql.catalog." + CATALOG_NAME, "org.apache.iceberg.spark.SparkCatalog");
+    configMap.put("spark.sql.catalog." + CATALOG_NAME + ".catalog-impl", "org.apache.iceberg.nessie.NessieCatalog");
+    configMap.put("spark.sql.catalog." + CATALOG_NAME + ".uri", this.uri);
+    configMap.put("spark.sql.catalog." + CATALOG_NAME + ".ref", this.ref);
+    configMap.put("spark.sql.catalog." + CATALOG_NAME + ".authentication.type", this.authenticationType);
+    configMap.put("spark.driver.extraJavaOptions", "-Dpackaging.type=jar -Djava.io.tmpdir=/tmp");
+
+    //adds all the S3 related configurations.
+    Preconditions.checkArgument(this.storageConfig instanceof S3Config, "Nessie Catalog only supports S3 file system.");
+    configMap.putAll(this.storageConfig.sparkConfigMap(CATALOG_NAME));
+    return configMap;
+  }
+
+  @Override
+  public Catalog genCatalog() {
+    NessieCatalog catalog = new NessieCatalog();
+    Map<String, String> properties = new HashMap<>(this.storageConfig.catalogInitializeProperties());
+    properties.put(CatalogProperties.URI, this.uri);
+    properties.put(CatalogProperties.WAREHOUSE_LOCATION, this.storageConfig.getWarehouseUri());
+    if (isNotBlank(this.ref)) {
+      properties.put("ref", this.ref);
+    }
+    if (isNotBlank(this.authenticationType)) {
+      properties.put(NESSIE_CATALOG_AUTHENTICATION_TYPE_KEY, this.authenticationType);
+    }
+    if (isNotBlank(this.nessieApiVersion)) {
+      properties.put("client-api-version", this.nessieApiVersion);
+    }
+    if (isNotBlank(this.token)) {
+      properties.put(CONF_NESSIE_AUTH_TOKEN, this.token);
```

Comment:
Nessie [supports](https://projectnessie.org/tools/client_config/) multiple authentication types. Setting just this token is sufficient for the `BEARER` auth. type, but `OAUTH2`, for example, requires more properties. Would it be possible to make [them](https://github.com/projectnessie/nessie/blob/nessie-0.74.0/api/client/src/main/java/org/projectnessie/client/NessieConfigConstants.java) configurable too?

Reply:
IMO, we should support other options with some UI changes like auth type based options. Else it will be many input fields on the form.

Reply:
I'm fine with supporting only `NONE` and `BEARER` auth types, but in that case I think we have to validate the auth type value and provide a clear error message in case something else is requested.

Reply:
Similarly, I think allowing the auth type to be set to any value, while we only support setting the bearer token (but not OAuth2 settings) can be confusing to end users.